src/Integration.Vsix/SonarLintTagger/AccumulatingIssueConsumer.cs

```diff
@@ -0,0 +1,117 @@
+ * passed to the OnIssuesChanged delegate.
+ * 
+ * However, it's possible that the text buffer could have been edited since the analysis
+ * was triggered. It is the responsibity of the callback to translate to supplied IssueMarkers
+ * to the current snapshot, if necessary.
+ * 
+ * 
+ * Mapping from reported issue line/char positions to the analysis snapshot
+ * ------------------------------------------------------------------------
+ * Currently, the JS/CFamily analyzers run against the file on disc, not the content of 
+ * the snapshot. We're making a couple of assumptions:
+ * 1) we're triggering analysis on save, so the state of the text buffer matches the file on disc, and
+ * 2) the file won't have had time to change again before the analyzer reads it so we expect always to
+ *    be able to map back from the reported issue to the snapshot.
+ * However, we code defensively and strip out any issues that can't be mapped to the analysis snapshot.
+ * 
+ */
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    /// <summary>
+    /// Handles processing the issues for a single analysis request
+    /// </summary>
+    internal class AccumulatingIssueConsumer : IIssueConsumer
+    {
+        // See bug #1487: this text snapshot should match the content of the file being analysed
+        private readonly ITextSnapshot analysisSnapshot;
+        private readonly IIssueToIssueMarkerConverter issueMarkerConverter;
+        private readonly List<IssueMarker> allIssueMarkers;
+        private readonly string analysisFilePath;
+        private readonly OnIssuesChanged onIssuesChanged;
+
+        public delegate void OnIssuesChanged(IEnumerable<IssueMarker> issueMarkers);
+
+        public AccumulatingIssueConsumer(ITextSnapshot analysisSnapshot, string analysisFilePath, OnIssuesChanged onIssuesChangedCallback)
+            : this(analysisSnapshot, analysisFilePath, onIssuesChangedCallback, new IssueToIssueMarkerConverter())
+        {
+        }
+
+        internal /* for testing */ AccumulatingIssueConsumer(ITextSnapshot analysisSnapshot, string analysisFilePath, OnIssuesChanged onIssuesChangedCallback,
+            IIssueToIssueMarkerConverter issueMarkerConverter)
+        {
+            this.analysisSnapshot = analysisSnapshot ?? throw new ArgumentNullException(nameof(analysisSnapshot));
+            this.analysisFilePath = analysisFilePath ?? throw new ArgumentNullException(nameof(analysisFilePath));
+            this.onIssuesChanged = onIssuesChangedCallback ?? throw new ArgumentNullException(nameof(onIssuesChangedCallback));
+
+            this.issueMarkerConverter = issueMarkerConverter ?? throw new ArgumentNullException(nameof(analysisSnapshot));
+
+            allIssueMarkers = new List<IssueMarker>();
+        }
+
+        public void Accept(string path, IEnumerable<IAnalysisIssue> issues)
+        {
+            // Callback from the daemon when new results are available
+            if (path != analysisFilePath)
```

Comment:
Unrelated: We should change the signature of IIssueConsumer. The 'path' parameter is redundant: each issue has a filePath, so if anything, we should be testing that all issues are applicable analysisFilePath.

Reply:
Yep, it is odd. I considered changing it, but as you say it's unrelated so I left it for another day.